conjure-java-undertow-runtime/src/test/java/com/palantir/conjure/java/undertow/runtime/ClientDisconnectTest.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.github.stefanbirkner.systemlambda.SystemLambda;
+import com.google.common.util.concurrent.Uninterruptibles;
+import io.undertow.Undertow;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.time.Duration;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class ClientDisconnectTest {
+
+    private Undertow server;
+    private CountDownLatch latch = new CountDownLatch(1);
+
+    @BeforeEach
+    public void before() {
+        server = Undertow.builder()
+                .addHttpListener(12345, "localhost")
+                .setHandler(new ConjureExceptionHandler(
+                        _exchange -> Uninterruptibles.sleepUninterruptibly(Duration.ofMillis(50)),
+                        (_e, _t) -> latch.countDown()))
```

Comment:
This doesnt work, its not actually concurrent so the client isn't disconnecting while the handler is working. WIP

Reply:
The default exception handler implementation logs the line we're looking for. We still must delegate to that exception handler prior to counting down the latch.

Reply:
We still need to use the `ConjureHandler` and register a custom wrapping `ExceptionHandler` with that conjure handler in order to use the server-side async codepath

Reply:
Thanks for the pointers!